go/vt/mysqlctl/builtinbackupengine.go

```diff
@@ -223,18 +223,11 @@ func (be *BuiltinBackupEngine) ExecuteBackup(ctx context.Context, params BackupP
 
 // getIncrementalFromPosGTIDSet turns the given string into a valid Mysql56GTIDSet
 func getIncrementalFromPosGTIDSet(incrementalFromPos string) (replication.Mysql56GTIDSet, error) {
-	pos, err := replication.DecodePositionDefaultFlavor(incrementalFromPos, replication.Mysql56FlavorID)
+	_, gtidSet, err := replication.DecodePositionMySQL56(incrementalFromPos)
 	if err != nil {
 		return nil, vterrors.Wrapf(err, "cannot decode position in incremental backup: %v", incrementalFromPos)
 	}
-	if !pos.MatchesFlavor(replication.Mysql56FlavorID) {
-		return nil, vterrors.Errorf(vtrpc.Code_FAILED_PRECONDITION, "incremental backup only supports MySQL GTID positions. Got: %v", incrementalFromPos)
```

Comment:
IMO it would be nice to move this wording to the error above so that it's clear that we don't support other flavors.

Reply:
This is likewise an intentional design: we `Wrapf` an error, which happens to be `ErrExpectMysql56Flavor`, and which explicitly tells us `"expected MySQL GTID position but found a different or invalid format"`. It would be redundant to add the same wording here.

Reply:
Right, but we lose the wording which notes that MySQL format is the only one supported. That was my point. I guess it's implicit in this wording, so not a huge deal.

Reply:
We don't lose the wording. The error above will have both `cannot decode position in incremental backup: %v` as well as `"expected MySQL GTID position but found a different or invalid format`, because it wraps one with another. Am I misunderstanding?